jsonrpc/jsonrpc.go

```diff
@@ -220,7 +225,14 @@ func (j *JSONRPC) handle(w http.ResponseWriter, req *http.Request) {
 
 	resp, err := j.dispatcher.Handle(data)
 	if err != nil {
-		handleErr(err)
+		id := err.(*ErrorObject).Data
+		err.(*ErrorObject).Data = nil
+		errorResponse := ErrorResponse{
+			JSONRPC: "2.0",
+			ID:      id,
+			Error:   err.(*ErrorObject),
+		}
```

Comment:
This `err` is always the value of *ErrorObject? If not, you might want to add some check here. or we should consider to let `Handle` return *ErrorObject.